swarm/pss/pss_test.go

```diff
@@ -681,21 +814,60 @@ func benchmarkSymkeyBruteforceSameaddr(b *testing.B) {
 	if err != nil {
 		b.Fatalf("could not generate whisper envelope: %v", err)
 	}
-	ps.Register(&topic, func(msg []byte, p *p2p.Peer, addr []byte) error {
+	ps.Register(&topic, func(msg []byte, p *p2p.Peer, asymmetric bool, keyid string) error {
 		return nil
 	})
 	pssmsg := &PssMsg{
-		To:      potaddr[len(potaddr)-1][:],
+		To:      addr[len(addr)-1][:],
 		Payload: env,
 	}
 	for i := 0; i < b.N; i++ {
-		err := ps.Process(pssmsg)
+		err := ps.process(pssmsg)
 		if err != nil {
 			b.Fatalf("pss processing failed: %v", err)
 		}
 	}
 }
 
+func setupNetwork(numnodes int) (clients []*rpc.Client, err error) {
+	nodes := make([]*simulations.Node, numnodes)
+	clients = make([]*rpc.Client, numnodes)
+	if numnodes < 2 {
+		return nil, fmt.Errorf("Minimum two nodes in network")
+	}
+	adapter := adapters.NewSimAdapter(services)
+	net := simulations.NewNetwork(adapter, &simulations.NetworkConfig{
+		ID:             "0",
+		DefaultService: "bzz",
+	})
```

Comment:
The network should be shutdown either if there is an error in `setupNetwork` or when the surrounding test finishes.

Reply:
`network.Shutdown()` works now?